### PR TITLE
Create a new component for a Card

### PR DESCRIPTION
**Pull request by ncounter**
The new **Card** is a [View Component](https://viewcomponent.org/) meant to unify usages of the [Boostrap card](https://getbootstrap.com/docs/5.0/components/card/) when it is a simple element with a title, few actions and a custom content. The implementation uses advantages of [View Component Slots](https://viewcomponent.org/guide/slots.html) and it comes with some [View Component Previews](https://viewcomponent.org/guide/previews.html).

The previews to check out how the component can look like are available [here](https://obs-reviewlab.opensuse.org/ncounter-card-component/rails/view_components/card_component/)

### CODE REVIEW

**Comment by ncounter** [comment]
[Unify Card Index Pattern (L)](https://trello.com/c/Ym4kvbUL/1872-unify-card-index-pattern-l)

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [inline src/api/app/components/card_component.html.haml:6]: This is specific to how the view component would be used in the context of repositories. It doesn't apply to other contexts. You should rely on [slots](https://viewcomponent.org/guide/slots.html). You could have a slot for the card header, body and footer for example.

Comment (danidoni) [reply src/api/app/components/card_component.html.haml:6]: Oh, nice tip... We'll take a look at slots for sure

File: src/api/app/components/card_component.html.haml
Comment (ncounter) [reply src/api/app/components/card_component.html.haml]: Sounds a good idea. I addressed that. There is one "drawback" about it, that one should remember to pass action links with the `nav-link` class, but we can live with it. :+1:

File: src/api/app/components/card_component.html.haml
Comment (saraycp) [inline src/api/app/components/card_component.html.haml:7]: Is it possible to check the presence of the `delete_button` so we don't introduce an empty `div`?

Comment (ncounter) [reply src/api/app/components/card_component.html.haml]: Good point, done. :+1:

**Review by saraycp — APPROVED** [review]

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [inline src/api/app/components/card_component.html.haml:8]: With this approach, we will have to duplicate the delete button over and over again. This could be streamlined to ensure the delete buttons are always consistent. This could be a component.

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [inline src/api/app/components/card_component.html.haml:1]: What is the reason for having a fixed width? Having the component adjust to its content would make it more flexible. Here's an example from a PR based on the card component, there's clearly too much whitespace. It could easily have half the width and still be fine.

whitespace

File: src/api/app/components/card_component.html.haml
Comment (ncounter) [reply src/api/app/components/card_component.html.haml:8]: That's the plan, to have a root component to reuse and do not _reinvent the wheel_ every time. I felt it was for a separate refactoring PR (as we do already have duplication around about it). Would you feel comfortable to keep it as it is for now, or do you prefer to add it here and now before merging the card component?

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [inline src/api/app/components/card_component.html.haml:2]: `.repository-card` should not be here, right? It shouldn't apply to all card components.

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [reply src/api/app/components/card_component.html.haml:1]: Could we not merge this line with the `.card` line below? Having an container for the `.card` container isn't needed.

File: src/api/app/components/card_component.html.haml
Comment (dmarcoux) [reply src/api/app/components/card_component.html.haml:8]: Handling this where the card is used, that would be a good first step. It doesn't have to happen in this PR though.

File: src/api/app/components/card_component.html.haml
Comment (ncounter) [reply src/api/app/components/card_component.html.haml:1]: Fixed width is fine I'd say, but not in a hard way. I mean, the card component should not have fixed width, I agree, I will adjust it. But it'd still be good if every usage of the card keeps in mind to set the width for the occurrences of that type, i.e.: all the repository cards in the repository page should have the same width, and all the token cards in the token page should have the same width as well, but the two width for repository and token can be different, that one can.

Anyway, I'm going to adapt here the card component to not set the width, :+1:

**Pull request merged**
